auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -378,6 +368,79 @@ public IdpConfig build() {
         }
     }
 
+    public static class PhoneIdpConfig extends IdpConfig {
```

Comment:
I feel like introducing a config per for phone auth provider leaves me desiring symmetry for such a config in other providers. 

Also, the type safety that this implementation gives is hard to enforce in a backward compatible way. This leads to unsafe typecasting like the one made in https://github.com/firebase/FirebaseUI-Android/blob/d1da32786f33f7a5552d3a3c5c778eb58c24305a/auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java#L118

My suggestion would be to introduce a Bundle parameter into the existing IdpConfig that yields an API that would look like:

Bundle b = new Bundle();
b.putString(AuthUI.PHONE_VERIFICATION_PROVIDER.DEFAULT_PHONE_EXTRA, "+16478488383");

new AuthUI.IdpConfig.Builder(AuthUI.PHONE_PROVIDER).setParams(bundle).build()) and wire it into the providers from that point on.

Reply:
@ashwinraghav Thhank's for the suggestion. I was thinking about extending existing IdpConfig but in that case were are adding changes to all configs (Google, Facebook) and they they don't need that parameter.

Anyway I think your solution is more elegant. I just want to propose to use List of Bundles instead of just Bundle:

AuthUI.IdpConfig.Builder(AuthUI.PHONE_PROVIDER).setParams(List<Bundle>).build())

In that case we can pass multiple params.
@ashwinraghav what do you think?